src/App.tsx

```diff
@@ -1,26 +1,241 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  USER_ID,
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { ErrorNotification } from './components/errorNotification';
+import { Footer } from './components/footer';
+import { Header } from './components/header';
+import { TodoList } from './components/todoList';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
-
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodo, setNewTodo] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [loading, setLoading] = useState<boolean>(false);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [editingTodoTitle, setEditingTodoTitle] = useState<string>('');
+
+  useEffect(() => {
+    if (USER_ID) {
+      setLoading(true);
+      getTodos()
+        .then(setTodos)
+        .catch(() => setError('Unable to load todos'))
+        .finally(() => setLoading(false));
+    }
+  }, []);
+
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!newTodo.trim()) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    setLoading(true);
+
+    const newTempTodo: Todo = {
+      id: 0,
+      title: newTodo,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo(newTempTodo);
+
+    createTodo(newTodo)
+      .then((newTodoItem: Todo) => {
+        setTodos(prevTodos => [...prevTodos, newTodoItem]);
+        setNewTodo('');
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+        setTempTodo(null);
```

Comment:
do it in finally instead of in both callbacks (then, catch)

Reply:
Not fixed:
![image](https://github.com/user-attachments/assets/8358be99-d273-4beb-b00a-8d39c6ab8ffb)